js/rank_states_utils.js

```diff
@@ -17,12 +17,20 @@ stateMap.append('g')
 svgStates.append('g')
   .attr('id','ranked-states-bars');
   
-var bardata = [{"wu":80, "abrv":'OK', "open": true}, {"wu":140, "abrv":'WI', "open":false}, {"wu":180, "abrv":'MI', "open":true}, 
-    {"wu":210, "abrv":'ID', "open":true}, {"wu":260, "abrv":'TX', "open":false}, {"wu":310, "abrv":'CA', "open":false}];
-
+// Read state data and add it to figure
+d3.queue()
+  .defer(d3.json, "data/wu_state_data.json")
+  .await(rankEm);
```

Comment:
This looks functional. But for your d3 edification, another option when you have just one data file is that instead of these 3 lines you can instead do
```js
d3.json('data/wu_state_data.json', rankEm);
```
where then `rankEm` would take a slightly different approach to arguments (they get passed as typical arguments rather than being contained within the `arguments` vector):
```js
rankEm(error, bardata) {
  if(error) throw error;
  svgStates.select....
}
```

Reply:
oooo I didn't realize that. Thought we had to nest or queue. Cool!